test/src/commonTest/kotlin/io/realm/LogTests.kt

```diff
@@ -0,0 +1,174 @@
+import io.realm.LogConfiguration
+import io.realm.Utils
+import io.realm.internal.RealmLog
+import io.realm.log.LogLevel
+import io.realm.util.TestLogger
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNull
+import kotlin.test.assertTrue
+
+/**
+ * Class that exercises the logger implementation.
+ *
+ * It isn't possible, in any sane manner, to validate the output actually being correct, so this class mostly just
+ * exercises the relevant code paths in order to check for crashes.
+ */
+class LogTests {
+
+    private lateinit var log: RealmLog
+
+    @BeforeTest
+    fun setUp() {
+        log = RealmLog(configuration = LogConfiguration(LogLevel.ALL, false, listOf()))
+    }
+
+    @Test
+    fun ignoreEventsLowerThanLogLevel() {
+        val customLogger = TestLogger()
+        log = RealmLog(configuration = LogConfiguration(LogLevel.WARN, true, listOf(customLogger)))
+        log.warn("Testing 1")
+        assertEquals("Testing 1", customLogger.message)
+        log.error("Testing 2")
+        assertEquals("Testing 2", customLogger.message)
+        log.info("Testing 3") // This should be swallowed
+        assertEquals("Testing 2", customLogger.message)
+    }
+
+    @Test
+    fun customLogger() {
+        val customLogger = TestLogger()
+        log = RealmLog(configuration = LogConfiguration(LogLevel.ALL, true, listOf(customLogger)))
+        var message = "Testing"
+
+        // Simple message
+        log.warn(message)
+        assertEquals(LogLevel.WARN, customLogger.logLevel)
+        assertNull(customLogger.throwable)
+        assertEquals(message, customLogger.message)
+        assertTrue(customLogger.args.isEmpty())
+
+        // Advanced message
+        val throwable = RuntimeException("BOOM")
+        message = "Message: %s"
+        val args: Array<out Any?> = arrayOf("foo")
+        log.error(throwable, message, *args)
+        assertEquals(LogLevel.ERROR, customLogger.logLevel)
+        assertEquals(throwable, customLogger.throwable)
+        assertEquals(message, customLogger.message)
+        assertTrue(customLogger.args.contentEquals(args))
+    }
+
+    @Test
+    fun smallLogEntry() {
```

Comment:
Is some part of the verification missing, or is it just to ensure that the codepaths works? If so, then maybe note it.

Reply:
There is a class comment here https://github.com/realm/realm-kotlin/pull/171/files#diff-8e3f6edc9620b29f9c42eeb34c2bdd2be307144396d4d0e55175ca585c34e9c7R13 

But I can replicate it to all the tests (about 5) if you think that would be better?

Reply:
why not doing something like https://github.com/realm/realm-kotlin/pull/171/files#diff-8e3f6edc9620b29f9c42eeb34c2bdd2be307144396d4d0e55175ca585c34e9c7R33 (i.e using a custom logger that logs to a StringBuffer or a file then make sure the content matches what you've logged?)